lib/gtfs.ex

```diff
@@ -73,16 +73,38 @@ defmodule Gtfs do
   end
 
   def handle_call({:timepoints_on_route, route_id}, _from, {:loaded, gtfs_data} = state) do
-    timepoint_ids =
+    route_patterns_by_direction =
       gtfs_data.route_patterns
       |> Enum.filter(fn route_pattern -> route_pattern.route_id == route_id end)
-      |> Enum.map(fn route_pattern -> route_pattern.representative_trip_id end)
-      |> Enum.flat_map(fn trip_id -> gtfs_data.stop_times[trip_id] end)
-      |> Enum.map(fn stop_time -> stop_time.timepoint_id end)
-      |> Enum.uniq()
-      |> List.delete("")
+      |> Enum.group_by(fn route_pattern -> route_pattern.direction_id end)
```

Comment:
might be a personal preference, but this can also be `Enum.group_by(& &1.direction_id)`

Reply:
It could. I have a personal preference for named arguments, since they give more of a hint of what's going on. That's not really necessary here, but since I switch between operating on route_patterns, directions, and stop_times in one function, I think it helps a bit.